dogfood/coder/Dockerfile

```diff
@@ -9,7 +9,7 @@ RUN cargo install exa bat ripgrep typos-cli watchexec-cli && \
 FROM ubuntu:jammy@sha256:0e5e4a57c2499249aafc3b40fcd541e9a456aab7296681a3994d631587203f97 AS go
 
 # Install Go manually, so that we can control the version
-ARG GO_VERSION=1.24.1
+ARG GO_VERSION=1.22.8
```

Comment:
Looks like we are on 1.22.12 in [some places](https://github.com/coder/coder/pull/17058), but only 1.22.9 in [others](https://github.com/coder/coder/blob/77fe10ead8e4077a7ba05b9f19dd778e8598690c/go.mod#L3)

Reply:
Maybe it makes more sense to roll forward at this stage instead?